src/main/java/StringAddCalculator.java

```diff
@@ -0,0 +1,61 @@
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class StringAddCalculator {
+    private static final String BASIC_DELIMITER = ",|:";
+    private static final int EMPTY_NULL_RETURN_VALUE = 0;
+
+
+    public static int calculateSum(String strValue) {
+        if (isEmptyOrNull(strValue))
+            return EMPTY_NULL_RETURN_VALUE;
+
+        return split(StringsToInts(getStringValues(strValue, getMatcher(strValue))));
+    }
+
+    private static String getDelimiter(Matcher m) {
+        if (m.find()) {
+            return BASIC_DELIMITER + "|" + m.group(1);
+        }
+        return BASIC_DELIMITER;
+    }
+
+    private static String[] getStringValues(String strValue, Matcher m) {
+        String delimiter = getDelimiter(getMatcher(strValue));
+        if (m.find()) {
+            return m.group(2).split(delimiter);
+        }
+        return strValue.split(delimiter);
+    }
+
+    private static Matcher getMatcher(String strValue) {
+        return Pattern.compile("//(.)\n(.*)").matcher(strValue);
+    }
```

Comment:
```suggestion
    private static String[] getStringValues(String strValue) {
        Matcher matcher = Pattern.compile("//(.)\n(.*)").matcher(strValue);
        if (matcher.find()) {
            String delimiter = BASIC_DELIMITER + "|" + matcher.group(1);
            return matcher.group(2).split(delimiter);
        }
        return strValue.split(BASIC_DELIMITER);
    }
```
위와 같은 기능의 코드인데요 
차근차근 살펴보겠습니다.

- `getStringValues` 메소드의 `Matcher m` 파라미터는 불필요해 보입니다.
  - 메소드에 넘겨 전달하지 않고도 함수 내부적으로 Matcher를 생성하면 되기 때문에 split할 문자열만 넘겨주면 될 것 같습니다.
  - 즉, getStringValues 메소드를 호출하는 쪽에서는 Matcher를 굳이 몰라도된다라는 의미입니다.
- `getMatcher` 메소드가 꼭 필요할까?
  -  메소드 분리는 재사용하기 위함도 있지만 의미전달의 목적으로도 사용되는데요. `Pattern.compile("//(.)\n(.*)").matcher(strValue)` 코드만으로도 메소드와 동일한 의미를 충분히 표현하고 있다고 생각합니다. 또한 지금 상황에서 2번 호출할 필요도 없어보입니다.
- `getDelimiter` 메소드
  - `matcher.find()` 분기를 사용하여 delimiter를 구하는데요. delimiter를 구하고 다시 돌아와 `matcher.find()` 분기를 타고 있습니다. 그렇다면 `matcher.find()` 조건에서 바로 delimiter를 구해도 되지 않을까? 하는 저의 의견입니다.
  - 이 부분은 저의 개인적인 의견입니다. 지금 코드도 좋습니다!

Reply:
좋은 의견 감사드립니다! 코드가 더 간결해지고 보기 훨씬 좋은 것 같습니다!
'메소드는 한가지 일만한다'와, '극단적인 메소드 분리'를 계속 생각하다보니, 머리가 그쪽으로 돌아간 것 같네요 ㅋㅋ
감사합니다~